### PR TITLE
Added FiveDaysWeather interface. its component and table

### PR DESCRIPTION
**Pull request by NoPanlc**
Added FiveDaysWeather interface, its component and table

### CODE REVIEW

File: weatherapplication.client/src/app/core/components/current-weather-form/current-weather-form.component.html [hunk 1]
```diff
@@ -1,24 +1,59 @@
...
+  <form *ngIf="form" [formGroup]="form">
+    <table class="table table-borderless">
+      <tr>
+        <th>City Name:</th>
+        <th>Your e-mail:</th>
+        <th>City Code:</th>
+        <th>State Code:</th>
+      </tr>
+      <tr>
+        <td>
```
Comment (Oleksandr-Ryabishchuk) [inline weatherapplication.client/src/app/core/components/current-weather-form/current-weather-form.component.html:11]: Empty lines should be removed

File: weatherapplication.client/src/app/core/components/five-days-weather-table/five-days-weather-table.component.html [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        <th>{{item.visibility}}</th>
+        <th>{{item.weatherDescription}}</th>
+        <th>{{item.weatherIcon}}</th>
+        <th>{{item.weatherMain}}</th>
+        <th>{{item.windDeg}}</th>
+        <th>{{item.windGust}}</th>
+        <th>{{item.windSpeed}}</th>
+      </tr>
+    </tbody>
+    <!--<thead>
```
Comment (Oleksandr-Ryabishchuk) [inline weatherapplication.client/src/app/core/components/five-days-weather-table/five-days-weather-table.component.html]: If we do not use comments - they should be removed

File: weatherapplication.client/src/index.html [hunk 3]
```diff
@@ -1,177 +1,297 @@
...
-          </li>
-        </ul>
+  <head>
+    <meta charset="utf-8" />
+    <title>Weather Application</title>
+    <base href="/" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <link rel="icon" type="image/x-icon" href="favicon.ico" />
+    <link
+      rel="stylesheet"
```
Comment (Oleksandr-Ryabishchuk) [inline weatherapplication.client/src/index.html:10]: Empty lines should be removed

File: weatherapplication.client/src/index.html [hunk 4]
```diff
@@ -1,177 +1,297 @@
...
+        <button
+          class="navbar-toggler"
+          type="button"
+          data-toggle="collapse"
+          data-target="#navbarSupportedContent"
+          aria-controls="navbarSupportedContent"
+          aria-expanded="false"
+          aria-label="Toggle navigation"
+        >
+          <span class="navbar-toggler-icon">
```
Comment (Oleksandr-Ryabishchuk) [inline weatherapplication.client/src/index.html:33]: Additional javascript/typescript code and additional html added to index.html should be refactored to a separate component called "applayout"

File: weatherapplication.client/src/index.html [hunk 5]
```diff
@@ -1,177 +1,297 @@
...
+    <link rel="icon" type="image/x-icon" href="favicon.ico" />
+    <link
+      rel="stylesheet"
+      href="https://cdn.jsdelivr.net/npm/bootstrap@4.6.0/dist/css/bootstrap.min.css"
+      integrity="sha384-B0vP5xmATw1+K9KRQjQERJvTumQW0nPEzvF6L/Z6nronJ3oUOFUFpCjEUQouq2+l"
+      crossorigin="anonymous"
+    />
+  </head>
+  <body>
+    <nav
```
Comment (Oleksandr-Ryabishchuk) [inline weatherapplication.client/src/index.html:17]: Additional HTML and javascript/typescript code should be refactored and put into a separate component called "applayout"

File: weatherapplication.client/src/index.html [hunk 6]
```diff
@@ -1,177 +1,297 @@
...
+            : 0.75,
+        loaderScreenTransition =
+          typeof window.loaderScreenTransition != "undefined"
+            ? window.loaderScreenTransition
+            : 0.675,
+        loaderScreenWidth =
+          typeof window.loaderScreenWidth != "undefined"
+            ? window.loaderScreenWidth
+            : 7.5;
+
```
Comment (Oleksandr-Ryabishchuk) [inline weatherapplication.client/src/index.html:118]: Most CSS Code from line 123 to line 285 should be converted from "style" to "classes" implementation and some of them should be used on conditionals like <div [class.example-class]="value === true"></div>

File: weatherapplication.client/src/index.html [hunk 7]
```diff
@@ -1,177 +1,297 @@
...
+            "px; justify-content: center; left: 0; margin: auto; max-height: 100% !important; max-width: 100% !important; min-height: 100vh; min-width: 100vh; position: fixed; top: 0; transition: " +
+            loaderScreenTransition +
+            "s ease-in-out; width: " +
+            innerWidth +
+            "px; z-index: 2147483647";
 
-    /* Logic
+          // Function
+          // Alpha
+          function alpha() {
```
Comment (Oleksandr-Ryabishchuk) [inline weatherapplication.client/src/index.html:172]: All the names for functions should have understandable names

File: weatherapplication.client/src/index.html [hunk 8]
```diff
@@ -1,177 +1,297 @@
...
+
+          // Test
+          function test() {
+            // Style > Data
+            // Background Color
+            data.style.backgroundColor = "transparent";
+
+            // Opacity
+            data.style.opacity = 0;
+
```
Comment (Oleksandr-Ryabishchuk) [inline weatherapplication.client/src/index.html:180]: The entire loader that we see when we move between pages should be moved to a separate component called "load-spinner" so we can give it time to exist. For example, we want to see "load-spinner" exactly 3 seconds. For this functionality, we don't need remove it from HTML manually, we need simple *ngIf and few variables

File: weatherapplication.client/src/app/core/components/five-days-weather-table/five-days-weather-table.component.html [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+        <th>Humidity</th>
+        <th>Difference between max and min temperature</th>
+        <th>Pop</th>
+        <th>Pressure</th>
+        <th>Rain</th>
+        <th>Sea Level</th>
+        <th>Snow</th>
+        <th>Temp</th>
+        <th>Temperature Max</th>
+        <th>Temperature Min</th>
```
Comment (yurikoles) [inline weatherapplication.client/src/app/core/components/five-days-weather-table/five-days-weather-table.component.html]: The same measure has two different name.

File: weatherapplication.client/src/app/app.module.ts [hunk 10]
```diff
@@ -19,18 +20,23 @@ import { FormsModule, ReactiveFormsModule } from '@angular/forms';
     CurrentWeatherPageComponent,
     FivedaysWeatherPageComponent,
```
Comment (yurikoles) [inline weatherapplication.client/src/app/app.module.ts:20]: I don't know such a word, `Fivedays`, seems to be a typo.

File: weatherapplication.client/src/app/app.module.ts [hunk 11]
```diff
@@ -19,18 +20,23 @@ import { FormsModule, ReactiveFormsModule } from '@angular/forms';
...
+    HttpClientModule,
     AppRoutingModule,
     RouterModule.forRoot([{ path: '**', component: RandomWeatherComponent }]),
     FormsModule,
     ReactiveFormsModule,
   ],
-  exports: [CurrentWeatherPageComponent, FivedaysWeatherPageComponent, StatisticsPageComponent],
+  exports: [
+    CurrentWeatherPageComponent,
+    FivedaysWeatherPageComponent,
```
Comment (yurikoles) [inline weatherapplication.client/src/app/app.module.ts:35]: Another typo.

File: weatherapplication.client/src/app/core/components/current-weather-form/current-weather-form.component.html [hunk 12]
```diff
@@ -1,24 +1,59 @@
...
+      </tr>
+      <tr>
+        <td></td>
+        <td></td>
+        <td></td>
+        <td>
+          <button
+            type="submit"
+            class="btn"
+            (click)="onSubmit()"
```
Comment (yurikoles) [inline weatherapplication.client/src/app/core/components/current-weather-form/current-weather-form.component.html]: Too generic name for a function.

File: weatherapplication.client/src/app/core/pages/fivedays-weather-page/fivedays-weather-page.component.ts [hunk 13]
```diff
@@ -1,10 +1,29 @@
...
+import { Component} from '@angular/core';
+import {
+  CurrentWeatherQuery,
+  FiveDaysWeather,
+  HttpService,
+} from 'src/app/services/http.service';
 
 @Component({
-  selector: 'app-fivedays-weather-page',
+  selector: 'fivedays-weather-page',
```
Comment (yurikoles) [inline weatherapplication.client/src/app/core/pages/fivedays-weather-page/fivedays-weather-page.component.ts:9]: The `fivedays` word again.

File: weatherapplication.client/src/app/core/pages/fivedays-weather-page/fivedays-weather-page.component.ts [hunk 14]
```diff
@@ -1,10 +1,29 @@
...
   templateUrl: './fivedays-weather-page.component.html',
-  styleUrls: ['./fivedays-weather-page.component.css']
+  styleUrls: ['./fivedays-weather-page.component.css'],
```
Comment (yurikoles) [inline weatherapplication.client/src/app/core/pages/fivedays-weather-page/fivedays-weather-page.component.ts:11]: A typo.

File: weatherapplication.client/src/app/core/pages/fivedays-weather-page/fivedays-weather-page.component.ts [hunk 15]
```diff
@@ -1,10 +1,29 @@
...
 })
 export class FivedaysWeatherPageComponent {
```
Comment (yurikoles) [inline weatherapplication.client/src/app/core/pages/fivedays-weather-page/fivedays-weather-page.component.ts:13]: Another typo.

File: weatherapplication.client/src/app/routing.module.ts [hunk 16]
```diff
@@ -1,23 +1,19 @@
...
-import { RouterModule, Routes } from "@angular/router";
-import { RandomWeatherComponent } from "./core/pages/random-weather/random-weather.component";
-import { CurrentWeatherPageComponent } from "./core/pages/current-weather-page/current-weather-page.component";
-import { FivedaysWeatherPageComponent } from "./core/pages/fivedays-weather-page/fivedays-weather-page.component";
-import { StatisticsPageComponent } from "./core/pages/statistics-page/statistics-page.component";
+import { NgModule } from '@angular/core';
+import { RouterModule, Routes } from '@angular/router';
+import { RandomWeatherComponent } from './core/pages/random-weather/random-weather.component';
+import { CurrentWeatherPageComponent } from './core/pages/current-weather-page/current-weather-page.component';
+import { FivedaysWeatherPageComponent } from './core/pages/fivedays-weather-page/fivedays-weather-page.component';
```
Comment (yurikoles) [inline weatherapplication.client/src/app/routing.module.ts:5]: And a typo again.

File: weatherapplication.client/src/app/routing.module.ts [hunk 17]
```diff
@@ -1,23 +1,19 @@
...
-      RouterModule.forRoot(routes),
-    ],
-    exports: [RouterModule]
-  })
-  export class AppRoutingModule { }
-  
\ No newline at end of file
+  { path: 'randomweather', component: RandomWeatherComponent },
+  { path: 'currentweather', component: CurrentWeatherPageComponent },
+  { path: 'fivedaysweather', component: FivedaysWeatherPageComponent },
```
Comment (yurikoles) [inline weatherapplication.client/src/app/routing.module.ts:11]: A typo.

File: weatherapplication.client/src/index.html [hunk 18]
```diff
@@ -1,177 +1,297 @@
...
-    if (condition())
-      // Update
-      update();
-
-    else {
-      // Initialization > Data
-      var data = setInterval(function () {
-        /* Logic
+      if (condition())
+        // Update
```
Comment (yurikoles) [inline weatherapplication.client/src/index.html]: Useless comments are worse than their absence. I see a lot of them in this file.

**Review by yurikoles — CHANGES_REQUESTED** [review]

File: weatherapplication.client/src/app/app.module.ts (hunk 10 above)
Comment (Oleksandr-Ryabishchuk) [reply weatherapplication.client/src/app/app.module.ts:21]: image

File: weatherapplication.client/src/app/core/pages/fivedays-weather-page/fivedays-weather-page.component.ts (hunk 13 above)
Comment (Oleksandr-Ryabishchuk) [reply weatherapplication.client/src/app/core/pages/fivedays-weather-page/fivedays-weather-page.component.ts:9]: image

File: weatherapplication.client/src/app/app.module.ts (hunk 11 above)
Comment (Oleksandr-Ryabishchuk) [reply weatherapplication.client/src/app/app.module.ts:35]: image

File: weatherapplication.client/src/app/core/pages/fivedays-weather-page/fivedays-weather-page.component.ts (hunk 14 above)
Comment (Oleksandr-Ryabishchuk) [reply weatherapplication.client/src/app/core/pages/fivedays-weather-page/fivedays-weather-page.component.ts:11]: image

File: weatherapplication.client/src/app/routing.module.ts (hunk 16 above)
Comment (Oleksandr-Ryabishchuk) [reply weatherapplication.client/src/app/routing.module.ts:5]: image

File: weatherapplication.client/src/app/routing.module.ts (hunk 17 above)
Comment (Oleksandr-Ryabishchuk) [reply weatherapplication.client/src/app/routing.module.ts:11]: image

**Pull request merged**